antarest/study/storage/study_upgrader/upgrader_870.py

```diff
@@ -50,10 +50,15 @@ def upgrade_870(study_path: Path) -> None:
 
     # Add properties for thermal clusters in .ini file
     ini_files = study_path.glob("input/thermal/clusters/*/list.ini")
+    thermal_path = study_path / Path("input/thermal/series")
     for ini_file_path in ini_files:
         data = IniReader().read(ini_file_path)
-        for section in data:
-            data[section]["costgeneration"] = "SetManually"
-            data[section]["efficiency"] = 100
-            data[section]["variableomcost"] = 0
+        area_id = ini_file_path.parent.name
+        for cluster in data.keys():
+            new_thermal_path = thermal_path / area_id / cluster.lower()
+            (new_thermal_path / "CO2Cost.txt").touch()
+            (new_thermal_path / "fuelCost.txt").touch()
```

Comment:
Do we have existing v8.7 studies in prod ? If so do you handle the empty state for those 2 matrices on existing studies ?

Reply:
We do not have 8.7 studies in prod